tech_specs/subprojects.md

```diff
@@ -0,0 +1,60 @@
+# Project division
+
+**Note:** See `service_components.txt` for diagrams.
+
+Current state of affairs is that all relevant code is spread over a set of repositories, contained in various accounts.
+
+It needs a more organized setup, but there are some considerations.
+* Functionality is logically spread over multiple loosely coupled parts:
+    * The controller firmware (C++)
+    * The controller <-> service communication layer - multiple variants (Python)
+    * (Future) plugins for external hardware not routed through the controller (Python)
+    * Various service-side functionality such as object / measurement databases (Pythyon)
+    * The Server-Side Events API (Python)
+    * The REST API (Python)
+    * The GUI (HTML/JS using the Vue/Vuex framework)
+    * (Future) Application wrappers around the GUI, making it a mobile or desktop app.
+* The long-term vision for the product includes it evolving into a generic industrial controller - without a brewery in sight.
```

Comment:
I would rephrase this into: keeping the platform open to other domains besides brewing. Do not get brewing specifics in layers that can stay generic.

Reply:
Fixed